src/mono/mono/metadata/icall-def.h

```diff
@@ -221,59 +221,59 @@ HANDLES(STREAM_1, "HasOverriddenBeginEndRead", ves_icall_System_IO_Stream_HasOve
 HANDLES(STREAM_2, "HasOverriddenBeginEndWrite", ves_icall_System_IO_Stream_HasOverriddenBeginEndWrite, MonoBoolean, 1, (MonoObject))
 
 ICALL_TYPE(MATH, "System.Math", MATH_1)
-NOHANDLES(ICALL(MATH_1, "Acos", ves_icall_System_Math_Acos))
-NOHANDLES(ICALL(MATH_1a, "Acosh", ves_icall_System_Math_Acosh))
-NOHANDLES(ICALL(MATH_2, "Asin", ves_icall_System_Math_Asin))
-NOHANDLES(ICALL(MATH_2a, "Asinh", ves_icall_System_Math_Asinh))
-NOHANDLES(ICALL(MATH_3, "Atan", ves_icall_System_Math_Atan))
-NOHANDLES(ICALL(MATH_4, "Atan2", ves_icall_System_Math_Atan2))
-NOHANDLES(ICALL(MATH_4a, "Atanh", ves_icall_System_Math_Atanh))
-NOHANDLES(ICALL(MATH_4b, "Cbrt", ves_icall_System_Math_Cbrt))
-NOHANDLES(ICALL(MATH_21, "Ceiling", ves_icall_System_Math_Ceiling))
-NOHANDLES(ICALL(MATH_5, "Cos", ves_icall_System_Math_Cos))
-NOHANDLES(ICALL(MATH_6, "Cosh", ves_icall_System_Math_Cosh))
-NOHANDLES(ICALL(MATH_7, "Exp", ves_icall_System_Math_Exp))
-NOHANDLES(ICALL(MATH_7a, "FMod", ves_icall_System_Math_FMod))
-NOHANDLES(ICALL(MATH_8, "Floor", ves_icall_System_Math_Floor))
-NOHANDLES(ICALL(MATH_22, "FusedMultiplyAdd", ves_icall_System_Math_FusedMultiplyAdd))
-NOHANDLES(ICALL(MATH_9, "Log", ves_icall_System_Math_Log))
-NOHANDLES(ICALL(MATH_10, "Log10", ves_icall_System_Math_Log10))
-NOHANDLES(ICALL(MATH_24, "Log2", ves_icall_System_Math_Log2))
-NOHANDLES(ICALL(MATH_10a, "ModF", ves_icall_System_Math_ModF))
-NOHANDLES(ICALL(MATH_11, "Pow", ves_icall_System_Math_Pow))
-NOHANDLES(ICALL(MATH_12, "Round", ves_icall_System_Math_Round))
-NOHANDLES(ICALL(MATH_14, "Sin", ves_icall_System_Math_Sin))
-NOHANDLES(ICALL(MATH_15, "Sinh", ves_icall_System_Math_Sinh))
-NOHANDLES(ICALL(MATH_16, "Sqrt", ves_icall_System_Math_Sqrt))
-NOHANDLES(ICALL(MATH_17, "Tan", ves_icall_System_Math_Tan))
-NOHANDLES(ICALL(MATH_18, "Tanh", ves_icall_System_Math_Tanh))
+NOHANDLES_FLAGS(ICALL(MATH_1, "Acos", ves_icall_System_Math_Acos), MONO_ICALL_FLAGS_NO_EXCEPTION)
```

Comment:
Should we go all the way to use MONO_ICALL_FLAGS_NO_EXCEPTION for math calls? Then it could also inform the direct icall linlining logic that the icall can be inlined (since that will drop the handling of lmf).